sds/src/scanner/mod.rs

```diff
@@ -92,6 +93,11 @@ pub trait CompiledRuleDyn: Send + Sync {
     fn get_match_validation_type(&self) -> Option<&MatchValidationType>;
 
     fn get_internal_match_validation_type(&self) -> Option<&InternalMatchValidationType>;
+
+    fn process_scanner_config(
+        &self,
+        group_config: &mut AHashMap<TypeId, Box<dyn Any + Send + Sync>>,
```

Comment:
You could look into using an `Anymap` here, which is the same abstraction you are writing yourself,
https://crates.io/crates/anymap2

Reply:
Nice thanks for pointing this out. I will create a follow up PR https://datadoghq.atlassian.net/browse/SDS-612